File: ChecksumCheck/Helper/Checksum.vb
Imports System.IO
Imports System.Security.Cryptography
Imports System.Text
Imports ChecksumCheck.DamienG.Security.Cryptography

Public Class Checksum

    Public Function ComputeFile(Path As String, Type As HashType) As String
        Dim buff As StringBuilder = New StringBuilder

        Select Case Type
            Case HashType.MD5
                Dim md5 As MD5CryptoServiceProvider = New MD5CryptoServiceProvider
                Dim f As FileStream = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                f = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                md5.ComputeHash(f)
                Dim hash As Byte() = md5.Hash
                Dim hashByte As Byte
                For Each hashByte In hash
                    buff.Append(String.Format("{0:X2}", hashByte))
                Next
                f.Close()
            Case HashType.SHA1
                Dim md5 As SHA1CryptoServiceProvider = New SHA1CryptoServiceProvider
                Dim f As FileStream = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                f = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                md5.ComputeHash(f)
                Dim hash As Byte() = md5.Hash
                Dim hashByte As Byte
                For Each hashByte In hash
                    buff.Append(String.Format("{0:X2}", hashByte))
                Next
                f.Close()
            Case HashType.SHA256
                Dim md5 As SHA256CryptoServiceProvider = New SHA256CryptoServiceProvider
                Dim f As FileStream = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                f = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                md5.ComputeHash(f)
                Dim hash As Byte() = md5.Hash
                Dim hashByte As Byte
                For Each hashByte In hash
                    buff.Append(String.Format("{0:X2}", hashByte))
                Next
                f.Close()
            Case HashType.SHA512
                Dim md5 As SHA512CryptoServiceProvider = New SHA512CryptoServiceProvider
                Dim f As FileStream = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                f = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                md5.ComputeHash(f)
                Dim hash As Byte() = md5.Hash
                Dim hashByte As Byte
                For Each hashByte In hash
                    buff.Append(String.Format("{0:X2}", hashByte))
                Next
                f.Close()
            Case HashType.CRC32
                Dim crc32 As Crc32 = New Crc32
                Dim f As FileStream = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                f = New FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192)
                crc32.ComputeHash(f)
                Dim hash As Byte() = crc32.Hash
                Dim hashByte As Byte
                For Each hashByte In hash
                    buff.Append(String.Format("{0:X2}", hashByte))
                Next
        End Select

        Return buff.ToString
    End Function

End Class

File: ChecksumCheck.Cmdl/Helper/Checksum.cs
using ChecksumCheck.Cmdl.Helper;
using System.IO;
using System.Security.Cryptography;
using System.Text;

public class Checksum
{
    public string CheckFile(string Path, int Type)
    {
        StringBuilder buff = new StringBuilder();

        switch (Type)
        {
            case 1:
                {
                    MD5CryptoServiceProvider md5 = new MD5CryptoServiceProvider();
                    FileStream f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    md5.ComputeHash(f);
                    byte[] hash = md5.Hash;
                    //INSTANT C# NOTE: Commented this declaration since looping variables in 'foreach' loops are declared in the 'foreach' header in C#:
                    //					byte hashByte = 0;
                    foreach (byte hashByte in hash)
                    {
                        buff.Append(string.Format("{0:X2}", hashByte));
                    }
                    f.Close();
                    break;
                }
            case 2:
                {
                    SHA1CryptoServiceProvider md5 = new SHA1CryptoServiceProvider();
                    FileStream f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    md5.ComputeHash(f);
                    byte[] hash = md5.Hash;
                    //INSTANT C# NOTE: Commented this declaration since looping variables in 'foreach' loops are declared in the 'foreach' header in C#:
                    //					byte hashByte = 0;
                    foreach (byte hashByte in hash)
                    {
                        buff.Append(string.Format("{0:X2}", hashByte));
                    }
                    f.Close();
                    break;
                }
            case 3:
                {
                    SHA256CryptoServiceProvider md5 = new SHA256CryptoServiceProvider();
                    FileStream f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    md5.ComputeHash(f);
                    byte[] hash = md5.Hash;
                    //INSTANT C# NOTE: Commented this declaration since looping variables in 'foreach' loops are declared in the 'foreach' header in C#:
                    //					byte hashByte = 0;
                    foreach (byte hashByte in hash)
                    {
                        buff.Append(string.Format("{0:X2}", hashByte));
                    }
                    f.Close();
                    break;
                }
            case 4:
                {
                    SHA512CryptoServiceProvider md5 = new SHA512CryptoServiceProvider();
                    FileStream f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    md5.ComputeHash(f);
                    byte[] hash = md5.Hash;
                    //INSTANT C# NOTE: Commented this declaration since looping variables in 'foreach' loops are declared in the 'foreach' header in C#:
                    //					byte hashByte = 0;
                    foreach (byte hashByte in hash)
                    {
                        buff.Append(string.Format("{0:X2}", hashByte));
                    }
                    f.Close();
                    break;
                }
            case 5:
                {
                    Crc32 crc32 = new Crc32();
                    FileStream f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    f = new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.Read, 8192);
                    crc32.ComputeHash(f);
                    byte[] hash = crc32.Hash;
                    //INSTANT C# NOTE: Commented this declaration since looping variables in 'foreach' loops are declared in the 'foreach' header in C#:
                    //				byte hashByte = 0;
                    foreach (byte hashByte in hash)
                    {
                        buff.Append(string.Format("{0:X2}", hashByte));
                    }
                    //Dim crc32 As New Crc32
                    //Dim hash As String = ""
                    //Using fs As FileStream = File.Open(Path, FileMode.Open)
                    //    For Each b As Byte In crc32.ComputeHash(fs)
                    //        hash &= b.ToString("x2").ToLower()
                    //    Next
                    //End Using
                    //buff.Append(hash)
                    break;
                }
        }

        return buff.ToString().ToLower();
    }
}